### DISCUSSION TITLE
Where does znap function keeps its generated files?

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [marlonrichert/zsh-snap#186] [Q&A] [answered]
I'm quite sure that the code `znap function _pipenv_completion pipenv 'eval "$(_PIPENV_COMPLETE=zsh_source pipenv)"'` will create a file `_pipenv_completion` somwhere.

I've searched the whole of my user path but can't find the file. Mind you completion works ok for `pipenv` and all the other completions I've defined in the above manner.

According to the docs they are supposed to be found at `${XDG_DATA_HOME:-~/.local/share}/zsh/site-functions.` however on my system, they're not there.

All I have in that location are the completions generated by `znap fpath` and not `znap function`.

**Comment by marlonrichert** [comment]
As it says [in the Readme](https://github.com/marlonrichert/zsh-snap#automatic-cache-invalidation):

> You can delete them manually from $XDG_CACHE_HOME/zsh-snap/eval.

**Reply by Pilgrim1379** [reply]
I don't `znap function` generated files are stored in `$XDG_CACHE_HOME/zsh-snap/eval`. I have the following in my config:

```
znap fpath _cargo 'rustup  completions zsh cargo'
znap fpath _solana 'solana completion --shell zsh'
znap fpath _pdm 'pdm completion zsh'


znap function _pip_completion pip 'eval "$(pip completion --zsh)"'
compctl -K _pip_completion pip

znap function _pipx_completion pipx 'eval "$(register-python-argcomplete pipx)"'
compdef _pipx_completion pipx

znap function _pipenv_completion pipenv 'eval "$(_PIPENV_COMPLETE=zsh_source pipenv)"'
compdef _pipenv_completion pipenv


znap eval zcolors "zcolors ${(q)LS_COLORS}"    # Extra init code needed for zcolors.
znap eval zoxide 'zoxide init zsh'
```
The `znap fpath` generated files are in `~/.local/share/zsh/site-functions`.
The `znap eval` generated files are in `~/.cache/zsh-snap/eval`.

I can't locate where the `znap function` generated files are. My `$XDG_CACHE_HOME/zsh-snap/eval` folder only contains `zcolors.zsh` and `zoxide.zsh`.

**Reply by marlonrichert** [reply]
Oh, sorry, I read that wrong. I’ll post a new answer.

**Comment by marlonrichert** [comment, marked as answer]
`znap function` doesn’t generate any files. It simply creates a lazy-loading function in memory for a set of commands of which you want to defer initialization. 🙂 The function is discarded from memory after use.

**Reply by Pilgrim1379** [reply]
Ah that explains it!
